### DISCUSSION TITLE
Start and end times

### DISCUSSION THREAD

**Discussion by rodtucker** [amberelectric/public-api#52] [General]
I want to be able to download 12 hours of predictions from any particular time. I tried to insert a date and time (ISO 8601) in the startDate and endDate fields in a /sites{sitesId}/prices call, but the server returned an "Internal Server Error". I can get around the problem by downloading one or two days' worth of data (two if the call is after mid-day) and throwing away the unwanted data, but this seems wasteful. 

Is it possible to download prices from one date and time to another date and time?

**Comment by madpilot** [comment]
Are you looking specifically at forecasts? Then use `/sites/{siteid}/prices/current?next=24` - that will return the next 12 hours of forecasts.

**Comment by madpilot** [comment]
Note that you can't get predictions for arbitrary times, you can only get predictions that were made at the current time

**Comment by rodtucker** [comment]
Perfect. That will do it. Thanks.

**Comment by rodtucker** [comment]
I regularly download predictions from the API for the coming 15 hours. It works fine, but about once every 24 or 48 hours, the returned data covers only 2.5 hours rather than the 15 hours I have asked for.  This causes my python code to crash. I am devising a work-around, but I wonder why the API seems to have a small bug.

**Reply by madpilot** [reply]
Do you have examples of times and the query you are running? There should be more than 2.5 hours. 

If the problem is our end (That isn't a 500), it could be missing data from AEMO, so you might need to assume that data may be missing, but if you can give me the dates you are seeing, I can have a look

**Comment by rodtucker** [comment]
I think the problem might be at your end, because the json file was cut down as if I had used “?next=5”. There were fewer than usual entries for “general”, “controlled load”, and “feed-in”. It was not just a truncated json file.

 

The problem occurred at 1:26 am this morning. This is my query:

 

amber = requests.get('https://api.amber.com.au/v1/sites/XXXXXXXXXXXXXXXXXXXXXX/prices/current?next=29', headers={'accept': 'application/json', 'Authorization': 'Bearer psk_YYYYYYYYYYYYYYYYYYYYYY'})

**Reply by madpilot** [reply]
I'll have a look in the logs - Sounds like missing data from AEMO.

**Reply by madpilot** [reply]
I can't see anything obvious, so I'll have to set up some extra logging.

I'll pick this up on Monday.

**Comment by rodtucker** [comment]
I have now modified my code so that it tries again if there are any problems. It is no longer crashing, so no urgency.

**Comment by rodtucker** [comment]
The problem has popped up again. If I try to download the current data plus 39 more, I get a total of 34 data points, not 40 as expected. This is happening with my own software and also when I run it on the “For Developers” website. If you like, I can copy and past the output on the website if it helps.

 

 

 

From: Rod Tucker ***@***.*** ***@***.***> > 
Sent: Friday, 20 August 2021 10:43 AM
To: 'amberelectric/public-api' ***@***.*** ***@***.***> >
Subject: RE: [amberelectric/public-api] Start and end times (#52)

 

I think the problem might be at your end, because the json file was cut down as if I had used “?next=5”. There were fewer than usual entries for “general”, “controlled load”, and “feed-in”. It was not just a truncated json file.

 

The problem occurred at 1:26 am this morning. This is my query:

 

amber = requests.get('https://api.amber.com.au/v1/sites/XXXXXXXXXXXXXXXXXXXXXX/prices/current?next=29', headers={'accept': 'application/json', 'Authorization': 'Bearer psk_YYYYYYYYYYYYYYYYYYYYYY'})

**Reply by madpilot** [reply]
Ah, that's a different thing - around this time of the day, there are only 16 hours of forecasts from AEMO (so 34 intervals).

They add another 24 hours of data between 12 and 12:30.

You make a good point though - I'll add notes about this in the documentation
